### PR TITLE
Bug 1227459 - Allow using $local authentication method when executing…

### PR DESCRIPTION
**Pull request by lzoubek**
… CLI

scripts

Added new pluginConfig boolean property "Native Local Authentication". This
gets discovered to true only if $local authentication is the only way to
access native interface.

### CODE REVIEW

File: modules/plugins/jboss-as-7/src/test/java/org/rhq/modules/plugins/jbossas7/XmlFileReadingTest.java [hunk 1]
```diff
@@ -153,4 +153,18 @@ public void testGetRealm() throws Exception {
         assert "jboss.server.config.dir".equals(propsFilePathRel) : "Path was " + propsFileName;
     }
 
+    public void testIsLocalNativeAuth() throws Exception {
```
Comment (tsegismont) [inline modules/plugins/jboss-as-7/src/test/java/org/rhq/modules/plugins/jbossas7/XmlFileReadingTest.java]: Thanks for making this tested

**Comment by tsegismont** [comment]
This looks good to me. I think we should add some resource upgrade code as well. So that users don't have to go through a manual resource pluginconfig update process.

**Comment by tsegismont** [comment]
So you think resource upgrade is not a good idea?

**Comment by lzoubek** [comment]
I think it is a good idea. I squashed the commits before merge. Thank you for reviewing this.

**Comment by tsegismont** [comment]
Ok, it flew below my radar :smile:

**Pull request merged**